### PR TITLE
MRG Improvements to raw plot.  Related to issue #2204

### PR DESCRIPTION
**Pull request by jaeilepp**
Added scaling to raw plot and proj button is hidden if there are no projectors to apply. Also made it possible to scale up with '=' for mac users.

closes https://github.com/mne-tools/mne-python/issues/2204 https://github.com/mne-tools/mne-python/issues/2207

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Looks like a good start to me, can you try adding the code from #2207 while you're digging around in there?

**Comment by mainakjas** [comment]
I added the closes tag so that the issue gets automatically closed when this is PR is merged.

**Comment by mainakjas** [comment]
I'm not quite convinced that `-/+` should change @teonlamont . It's semantically more meaningful to me the way it is now.

File: mne/viz/epochs.py [hunk 1]
```diff
@@ -1056,7 +1056,7 @@ def _plot_onkey(event, params):
     elif event.key == '-':
         params['scale_factor'] /= 1.1
         _plot_traces(params)
-    elif event.key == '+':
+    elif event.key == '+' or event.key == '=':
```
Comment (mainakjas) [inline mne/viz/epochs.py]: Zen of python says: `There should be one-- and preferably only one --obvious way to do it.` :)

**Comment by teonlamont** [comment]
then this should be '=' with no '+'. it's wholly inconvenient. think of the user @mainakjas

**Comment by Eric89GXL** [comment]
@mainakjas I agree that there should be one way to do it, but if you think of `=` and `+` as being the same key, you can still make it fit :) Seriously though, it does seem to me like the added convenience of not holding shift is worth the deviation from Zen here.

@jaeilepp I see your latest commit, does that mean the fullscreen toggle actually worked for you?

**Comment by jaeilepp** [comment]
Yeah, it works for me.

**Comment by Eric89GXL** [comment]
Great, now we just need those pesky OSX users to try it and tell us it doesn't work on their backend :)

**Comment by agramfort** [comment]
works like a charm on my mac

+1 for merge

File: mne/viz/raw.py [hunk 2]
```diff
@@ -169,6 +169,17 @@ def _plot_raw_onkey(event, params):
     elif event.key in ['o', 'p']:
         _toggle_options(None, params)
         return
+    elif event.key == '+' or event.key == '=':
```
Comment (Eric89GXL) [inline mne/viz/raw.py]: BTW for future reference you can just do `elif event.key in ['+', '=']:`

**Comment by Eric89GXL** [comment]
+1 for merge even without my tweak, @jaeilepp I'll restart the Travis build until the memory error does not happen...

**Comment by Eric89GXL** [comment]
Travis is happy, @mainakjas or @dengemann feel free to have a final look and merge if you're happy

**Comment by Eric89GXL** [comment]
@jaeilepp are you done BTW? The title is still `WIP`, please change to `MRG` if you are indeed done.

**Comment by mainakjas** [comment]
I still can't change the number of channels in `raw.plot()`. Wasn't that supposed to be part of the issue I raised?

**Comment by mainakjas** [comment]
another minor inconsistency I noted (apologies @jaeilepp for being so nitpicky). The linewidth of the plot in `raw.plot()` and `epochs.plot()` look different to my eye. Can we make it consistent?

**Comment by jaeilepp** [comment]
Yeah, I was planning on implementing the channel adding/removing and adjusting the horizontal window size today.

**Comment by jaeilepp** [comment]
The lines looked different in width because antialiasing was set to false in raw.plot. Not sure why it was like that, but I removed it. I think it's now ready for review.

File: mne/io/base.py [hunk 3]
```diff
@@ -1287,9 +1287,12 @@ def plot(self, events=None, duration=10.0, start=0.0, n_channels=20,
         The arrow keys (up/down/left/right) can typically be used to navigate
         between channels and time ranges, but this depends on the backend
         matplotlib is configured to use (e.g., mpl.use('TkAgg') should work).
-        To mark or un-mark a channel as bad, click on the rather flat segments
-        of a channel's time series. The changes will be reflected immediately
-        in the raw object's ``raw.info['bads']`` entry.
+        The scaling can be adjusted with - and + or = keys. The viewport
```
Comment (teonlamont) [inline mne/io/base.py]: + or = --> + (or =) for consistency

File: mne/viz/raw.py [hunk 4]
```diff
@@ -361,10 +410,12 @@ def plot_raw(raw, events=None, duration=10.0, start=0.0, n_channels=None,
     -----
     The arrow keys (up/down/left/right) can typically be used to navigate
     between channels and time ranges, but this depends on the backend
-    matplotlib is configured to use (e.g., mpl.use('TkAgg') should work).
-    To mark or un-mark a channel as bad, click on the rather flat segments
-    of a channel's time series. The changes will be reflected immediately
-    in the raw object's ``raw.info['bads']`` entry.
+    matplotlib is configured to use (e.g., mpl.use('TkAgg') should work). The
+    scaling can be adjusted with - and + or = keys. The viewport dimensions can
```
Comment (teonlamont) [inline mne/viz/raw.py]: here too

**Comment by mainakjas** [comment]
LGTM now. Is the `=` sign thing going to be incorporated in the other PR now? Because that doesn't seem to work there ...

**Comment by mainakjas** [comment]
@dengemann please merge if you're happy

**Comment by Eric89GXL** [comment]
Please leave antialiasing off in the raw plotting. It is done for speed purposes. You might consider doing it for the epochs plotting, too, if you find that it is slow to redraw.

**Comment by Eric89GXL** [comment]
BTW if you ever wonder why something was done, you can use `git blame` to see who changed the line last, and then you can ask the person (in this case it should be me) and hope they remember the rationale :)

**Comment by dengemann** [comment]
Almost perfect. Unfortunately f11 collides with Mac OSX keyboard shortcuts ... so it does not work.

**Comment by dgwakeman** [comment]
using the ```fn``` key doesn't solve it?

**Comment by dengemann** [comment]
> using the fn key doesn't solve it?

nope.

**Comment by dengemann** [comment]
It does not even work when I deactivate the default mapping (show desktop). it does nothing on my box.

**Comment by dgwakeman** [comment]
@dengemann 
Have you tried these two changes in combination (or alone):
http://apple.stackexchange.com/questions/110525/how-do-i-get-f11-and-f12-to-behave-like-normal-function-keys

Note, you probably need to restart at least python if not the computer to get the remappings to work.

**Comment by dengemann** [comment]
> Note, you probably need to restart at least python if not the computer to get the remappings to work.

that's exactly what I did. The remapping btw. take place immedeately. It does not work, not even calling ` mng.full_screen_toggle()` directly

**Comment by Eric89GXL** [comment]
If calling that directly does not work, then it sounds like another limitation of the MacOSX mpl backend to me, and I'm not sure if there will be a good workaround. I guess you could try querying the screen size, and setting the window size to that and position to (0, 0), but this is likely to create more problems than it solves.

**Comment by dengemann** [comment]
So let's keep this open till we have robust solution.

2015-06-17 17:34 GMT+02:00 Eric Larson <notifications@github.com>:

**Comment by Eric89GXL** [comment]
I'm suggesting that might not happen until matplotlib updates their libraries, i.e. months or years, so I'm not sure that's an option.

**Comment by Eric89GXL** [comment]
My vote would be to have the functionality as it is now, and say that it might not work on the OSX backend. That way at least some people can have the functionality, and this PR isn't held up. Then OSX can be improved in a later PR.

**Comment by dengemann** [comment]
fair enough.

2015-06-17 20:55 GMT+02:00 Eric Larson <notifications@github.com>:

**Comment by agramfort** [comment]
fine with me too

**Comment by Eric89GXL** [comment]
Cool, I'll go ahead and merge then. Thanks for the tweaks @jaeilepp!

**Comment by mainakjas** [comment]
@jaeilepp I realized you forgot to update `whats_new` for this. Maybe you can squash it into one of your other PRs

**Pull request merged**